Client/Client.Qt/Widgets/ChatWidget.cpp

```diff
@@ -9,30 +11,37 @@ ChatWidget::ChatWidget(QWidget* parent) : QWidget(parent)
     _mainChatLayout = std::make_unique<QVBoxLayout>(this);
     _chatHistory    = std::make_unique<ChatHistory>(this);
     _textEdit       = std::make_unique<TextEdit>(_chatHistory.get());
+    _requestTimer   = std::make_unique<QTimer>();
 
     _mainChatLayout->setContentsMargins(0, 0, 0, 0);
     _mainChatLayout->setSpacing(0);
     _mainChatLayout->addWidget(_chatHistory.get(), 85);
     _mainChatLayout->addWidget(_textEdit.get(), 15);
     setLayout(_mainChatLayout.get());
 
+    connect(_requestTimer.get(), &QTimer::timeout, [=]() {
+        if (oApp->connectionManager()->isConnected())
+        {
+            oApp->connectionManager()->askForMessageHistory(_channelID);
+        }
+    });
+    _requestTimer->start(1000);
+
     setMinimumWidth(st::chatWidgetMinWidth);
     connect(_textEdit.get(), &TextEdit::sendMessage, this, &ChatWidget::newMessage);
     connect(ReceiverManager::instance(), &ReceiverManager::onMessageHistoryAnswer, this, &ChatWidget::addMessages);
 }
 
-void ChatWidget::newMessage(const QString& messageText)
-{
-    oApp->connectionManager()->storeMessage(messageText.toStdString(), _channelID);
-    _chatHistory->addMessage(messageText, QDateTime::currentSecsSinceEpoch());
-}
-
-void ChatWidget::newMessageA(const QString& messageText, const QString& username)
-{
-    _chatHistory->addMessage(messageText, QDateTime::currentSecsSinceEpoch(), username);
-}
+void ChatWidget::newMessage(const QString& messageText) { oApp->connectionManager()->storeMessage(messageText.toStdString(), _channelID); }
 
-void ChatWidget::addMessages(const std::vector<Network::MessageInfo>&)
+void ChatWidget::addMessages(const std::vector<Network::MessageInfo>& messages)
 {
-    /// TODO: implement it
+    for (const auto& message : messages)
+    {
+        if (message.channelID != _channelID)
+        {
+            continue;
+        }
+        _chatHistory->addMessage(message);
```

Comment:
? why not the next:
if (message.channelID == _channelID)
{
    addMessage
}

Reply:
okay